app/assets/javascripts/geoblacklight/viewers/esri/dynamic_map_layer.js

```diff
@@ -0,0 +1,85 @@
+//= require geoblacklight/viewers/esri
+
+GeoBlacklight.Viewer.DynamicMapLayer = GeoBlacklight.Viewer.Esri.extend({
+  layerID: null,
```

Comment:
```layerId``` is not an ESRI thing, however it is different from ```dataLayerId```. I'll rename it ```dynamicLayerId``` to avoid confusion.